backend/api_handler.py

```diff
@@ -74,11 +75,17 @@ def get_groups(claims):
         groups: list = (
             saml_groups.replace('[', '').replace(']', '').replace(', ', ',').split(',')
         )
-    cognito_groups = claims.get('cognito:groups', '').split(',')
-    groups.extend(cognito_groups)
+    cognito_groups = claims.get('cognito:groups', '')
+    if len(cognito_groups):
+        groups.extend(cognito_groups.split(','))
     return groups
 
 
+def get_custom_groups(user_id):
+    service_provider = ServiceProviderFactory.get_service_provider_instance()
+    return service_provider.get_groups_for_user(user_id)
+
```

Comment:
for 1. if using the custom auth, we can avoid calling this function. It still won't create any problems if we call it as it would return an empty list. But I will add code to avoid calling it. I was working on both cognito and okta and thus , missed adding conditions on calling the `get_groups` method. 
for 2. correct